printer_mods/bobbleheed/Filament_Runout/README.md

```diff
@@ -0,0 +1,57 @@
+# Filament Runout Sensor
+
+This mod replaces the stock bowden retainer with one of a similar profile that hides a runout sensor inside. 
+
+## BOM
+   | Material | Quantity |
+   |----------|----------|
+   | KW10 Microswitch | 1 |
+   | M2x8 SHCS | 2 |
+   | M3x8 SHCS | 3 |
+   | M3 Threaded Inserts | 2 |
+   | M3 Hammerhead T Nut | 1 |
+   | JST 2 OR 3 Pin Housings | 1 | 
+   | JST Contacts | 2 |
+   | Glue (CA or Hot) | | 
+   | Solder | |   
+<br>
+   
+### Note: 
+
+The KW10 microswitch can likely be swapped out for an Omron D2F-5L switch or other equivalents but this hasn't been tested. 
+
+Optionally, wires could be soldered directly to the microswitch negating the need for the JST terminal. 
+
+Also two of the M3x8 and the threade inserts are not required if you do not want to fit a cover over the switch.  
+<br>
+
+## Assembly
+
+![Assembly](./IMAGES/Internal_Assembly.png)
+
+Choose the appropriate housing for your printer, either left or right hand (LH or RH) depending on which side you run the reverse bowden to. Print using standard Voron print settings. 
+
+If you want to use the housing cover use a soldering iron to insert the threaded inserts into the locations as shown above. 
+
+Insert the switch into the housing with the lever opening facing the wider of the openings on the housing sides. Use the M2x8 to fix it in place. The switch should have clearance to fully open inside. 
+
+![Cross Section](./IMAGES/Switch_Cross_Section.png)
+
+Place the JST terminal inside close to the switch and affixing with glue, bridge the the terminals of the switch and JST with solder in the NC position. 
```

Comment:
Wouldn't the filament runout sensor be better in the NO position instead of NC?  If you connect it NC, and a wire breaks, it'll always read as filament being there and will never trigger when the filament runs out.  If you connect it NO, and a wire breaks, it'll read it as being out of filament so you'll notice something is wrong.  This is basically the reverse of endstop wiring, because endstop switch's normal state is not pressed in, but a filament switch's normal state is to be pressed in.

Reply:
You're correct about wiring in NO being better. I had been thinking of the endstop logic when writing the instructions.